src/main/scala/com/actian/spark_vector/datastream/DataStreamConnector.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2016 Actian Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.actian.spark_vector.datastream
+
+import java.net.InetSocketAddress
+import java.nio.channels.SocketChannel
+import java.nio.ByteBuffer
+
+import org.apache.spark.Logging
+
+import scala.reflect.ClassTag
+
+import com.actian.spark_vector.util.ResourceUtil.{ closeResourceOnFailure, closeResourceAfterUse }
+import com.actian.spark_vector.datastream.reader.DataStreamReader
+import com.actian.spark_vector.vector.ColumnMetadata
+import com.actian.spark_vector.srp.VectorSRPClient
+import com.actian.spark_vector.colbuffer.IntSize
+
+/**
+ * Container for the datastream connection header info.
+ *
+ * @note We keep just a part of this info, the rest of it such as column names and their logical/physical types
+ * is not stored in this container since we get it earlier from a JDBC query.
+ */
+private[datastream] case class DataStreamConnectionHeader(statusCode: Int,
+    numCols: Int,
+    vectorSize: Int,
+    private val colInfo: Array[(Boolean, Boolean)]) { // For now only isNullable and isConst
+  def isNullableCol(i: Int): Boolean = colInfo(i)._1
```

Comment:
these methods are more appropriate to the columns. So if you created a ColumnInfo(name: String, nullable: Boolean, const: Boolean) case class, this DataStreamConnectionHeader would contain a "cols: Seq[ColumnInfo]" and then to determine if the i-th column is nullable you could just do connectionHeader.cols(i).nullable. These methods won't be needed anymore.